### PR TITLE
JDK-8248904: Add support to jpackage for the Mac App Store

### PR DESCRIPTION
**Pull request by andyherrick**
Implementation of Mac App Support including three new mac specific CLI options.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8248904](https://bugs.openjdk.java.net/browse/JDK-8248904): Add support to jpackage for the Mac App Store 

### Reviewers
 * [Alexey Semenyuk](https://openjdk.java.net/census#asemenyuk) (@alexeysemenyukoracle - **Reviewer**)
 * [Alexander Matveev](https://openjdk.java.net/census#almatvee) (@sashamatveev - Committer)
 * [Alexander Zuev](https://openjdk.java.net/census#kizune) (@azuev-java - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - Author)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2716/head:pull/2716`
`$ git checkout pull/2716`

### CODE REVIEW

**Review by alexeysemenyukoracle — APPROVED** [review]

**Review by sashamatveev — APPROVED** [review]

**Review by azuev-java — APPROVED** [review]

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties [hunk 1]
```diff
@@ -185,7 +185,7 @@ Generic Options:\n\
 \          Icons, template files, and other resources of jpackage can be\n\
 \          over-ridden by adding replacement resources to this directory.\n\
 \          (absolute path or relative to the current directory)\n\
-\  --runtime-image <file-path>\n\
+\  --runtime-image <file path>\n\
```
Comment (kevinrushforth) [inline src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties:188]: This seems unrelated to this change.

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties [hunk 2]
```diff
@@ -241,8 +241,16 @@ MSG_Help_mac_launcher=\
 \          (absolute path or relative to the current directory).\n\
 \          If not specified, the standard keychains are used.\n\
 \  --mac-signing-key-user-name <team name>\n\
-\          Team name portion in Apple signing identities' names.\n\
-\          For example "Developer ID Application: "\n\
+\          Team or user name portion of Apple signing identities.\n\
+\  --mac-app-store\n\
+\          Indicates that the jpackage output is intended for the\n\
+\          Mac App Store.\n\
```
Comment (kevinrushforth) [inline src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties:247]: Maybe `macOS App Store`?

**Review by kevinrushforth — APPROVED** [review]
Looks good with a couple questions. Is `JavaApp.icns` intended to be an empty file (I see that the file it was renamed from was empty, so probably OK)? The rest are inline below.

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties (hunk 1 above)
Comment (andyherrick) [reply src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties:188]: yes - it is unrelated, other than I was adding a new help text with the same syntax (--mac-entitlements <file path>) and was looking for consistent way to say "<file path>" and found that in all cases except this one we said <file path> not <file-path> so I fixed this one too to make them all consistent.

File: src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties (hunk 2 above)
Comment (andyherrick) [reply src/jdk.jpackage/share/classes/jdk/jpackage/internal/resources/HelpResources.properties:247]: Although I do see it referenced both ways (even by Apple) I see 10 times as many entries in Google for "Mac App Store" (1.7 billion) vs MacOS App Store (155 million)
